tests/distributed/compose.yaml

```diff
@@ -0,0 +1,89 @@
+version: '3.7'
+
+services:
+
+  caddy_server:
+    image: caddy-docker-proxy:local
```

Comment:
@francislavoie @mholt
Is there any simple way to run easily run official docker image with Admin API exposed to *:2019?
Maybe something like `caddy admin-api --bind ":2019"`.

If it is possible, I will consider reducing this project to be only a caddy server controller, that discovers all caddy servers inside docker cluster and configures them.

Reply:
Hmm, that's a good question. I don't think we have anything in place for that now other than having an initial config as a base. Like make sure the initial Caddyfile has the admin global option set to `0.0.0.0:2019`.

I think we could add an env variable that would set the admin address (if it's not already set in a config file, in which case the config file would win).

Reply:
First, are you sure that's a good idea? That means any client on any interface can read and change the config.

But, to answer your question, you can change the admin endpoint by setting the `listen` property of the `admin` struct in your config: https://caddyserver.com/docs/json/admin/#listen

There's no command line flag for this since all config is in the config file. The only thing command line flags are for is bootstrapping the config or customizing the command itself. So you can use the `--config` flag to pass a config, even if it only sets that one field, then it will use that value at startup as if it was specified on the command line directly.

Reply:
@mholt it's fine to do it in this case because of Docker networking. Only containers in the same network could access it.